dbt/adapters/impala/connections.py

```diff
@@ -281,6 +286,32 @@ def close(cls, connection):
         except Exception as err:
             logger.debug(f"Error closing connection {err}")
 
+    @classmethod
+    def fetch_impala_version(cls, connection):
+
+        if ImpalaConnectionManager.impala_version: 
+            return ImpalaConnectionManager.impala_version
+
+        try:
+            sql = "select version()"
+            cursor = connection.cursor()
+            cursor.execute(sql)
+
+            res = cursor.fetchall()
+
+            ImpalaConnectionManager.impala_version = res[0][0].split("RELEASE")[0].strip()
+
+            tracker.populate_warehouse_info({ "version": ImpalaConnectionManager.impala_version, "build": res[0][0] })
+        except Exception as ex:
+            # we couldn't get the spark warehouse version, default to version 2
+            logger.debug(f"Cannot get impala version. Error: {ex}")
+            ImpalaConnectionManager.impala_version = "NA"
+
+            tracker.populate_warehouse_info({ "version": ImpalaConnectionManager.impala_version, "build": "NA" })
+
+        os.environ['DBT_IMPALA_VERSION'] = ImpalaConnectionManager.impala_version 
```

Comment:
Why do we need to set the env variable DBT_IMPALA_VERSION

Reply:
removed this.